### PR TITLE
add nominations from heltour to chesster

### PR DESCRIPTION
**Pull request by endrawes0**

### CODE REVIEW

File: src/heltour.js [hunk 1]
```diff
@@ -94,7 +94,26 @@ function updatePairing(heltourConfig, result) {
...
 
+function getPrivateURL(heltourConfig, page, user){
+    var options = url.parse(heltourConfig.base_endpoint + "get_private_url/");
+    options.parameters = {
+        'league': heltourConfig.league_tag,
+        'page': page,
+        'user': user
+    };
+
+    options.headers = {
```
Comment (lakinwecker) [inline src/heltour.js:105]: It's probably time to refactor this + the first line in all of these functions.

File: src/heltour.js [hunk 2]
```diff
@@ -94,7 +94,26 @@ function updatePairing(heltourConfig, result) {
...
+        'Authorization': 'Token ' + heltourConfig.token
+    };
+    return http.fetchURLIntoJSON(options).then(function(response){
+        if(response["error"]){
```
Comment (lakinwecker) [inline src/heltour.js]: I think this needs to be response['json']['error'] right?

File: src/heltour.js (hunk 1 above)
Comment (endrawes0) [reply src/heltour.js:105]: agreed

File: src/heltour.js [hunk 3]
```diff
@@ -94,7 +94,26 @@ function updatePairing(heltourConfig, result) {
...
+            throw new Error("error test: " + response["error"]);
+        }
+        return response;
```
Comment (lakinwecker) [inline src/heltour.js]: I would return response['json'] here.

File: src/heltour.js (hunk 2 above)
Comment (endrawes0) [reply src/heltour.js]: yeah.

File: src/chesster.js [hunk 4]
```diff
@@ -119,6 +119,36 @@ function leagueDMResponse(patterns, responseName) {
...
+        return;
+    }
+
+    var speaker = users.getByNameOrID(message.user);
+    heltour.getPrivateURL(
+        heltourOptions,
+        'nominate', //hardcoding this for now
+        speaker.name
+    ).then(function(getPrivateURLResult){
+        var json = getPrivateURLResult.json;
```
Comment (lakinwecker) [inline src/chesster.js]: If you return response['json'] from the previous method then you don't need this line here, you get the json object directly. Makes this code a bit more simple.

File: src/heltour.js (hunk 2 above)
Comment (endrawes0) [reply src/heltour.js]: and probably dont need the error test

File: src/chesster.js (hunk 4 above)
Comment (endrawes0) [reply src/chesster.js]: great suggestion

File: src/chesster.js [hunk 5]
```diff
@@ -119,6 +119,36 @@ function leagueDMResponse(patterns, responseName) {
...
 leagueDMResponse(['captains', 'captain list'], 'formatCaptainsResponse');
 
+/* game nomination */
+chesster.hears({
+    middleware: [ slack.requiresLeague ],
+    patterns: [ 'nomination' ],
+    messageTypes: [ 'direct_message' ]
+},
+function(bot, message) {
+    if (!message.league) {
```
Comment (lakinwecker) [inline src/chesster.js]: I don't think this is necessary if you use slack.requiresLeague, but it doesn't hurt either.

File: src/chesster.js [hunk 6]
```diff
@@ -182,6 +212,7 @@ function prepareCommandsMessage(){
         "    [ registration | sign up ]     ! registration form to play in our league\n" +
         "    [ source ]                     ! github repo for Chesster \n" +
         "    [ subscription help ]          ! help for chesster's subscription system\n" +
+        "    [ nomination <league> ]   ! get a private nominaion link for <league>, {45|lonewolf}, of your choosing\n" +
```
Comment (lakinwecker) [inline src/chesster.js]: typo. Nomina**t**ion

**Comment by lakinwecker** [comment]
Great work

**Pull request merged**